File: 3. Longest Substring Without Repeating Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        sLen = len(s)
        if sLen <= 1: return sLen  # s is shorter than 2 letters
        subStrLen, p1, p2 = 0, 0, 1  # use 2 pointers
        while p2 < sLen and subStrLen < 95:
            for p2 in xrange(p1, sLen):
                if s[p2] in s[p1:p2]:
                    break
                else:
                    p2 += 1
            subStrLen = max(p2 - p1, subStrLen) 
            p1 += 1
        return subStrLen

File: 3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int sLen = s.size(), subStrLen = 0, p1 = 0, p2 = 1;
        if (sLen <= 1)
            return sLen;
        string subStr;
        while (p2 < sLen && subStrLen < 95){
            for (p2 = p1; p2 < sLen; p2++){
                subStr.assign(s, p1, p2 - p1); //substr is s[p1:p2]
                if (subStr.find(s[p2]) != -1) // s[p2] in substr
                    break;
            }
            subStrLen = max(p2 - p1, subStrLen);
            p1++;
        }
        return subStrLen;
    };
};
